NCS.DSS.Customer/Cosmos/Provider/CosmosDBProvider.cs

```diff
@@ -0,0 +1,282 @@
+﻿using Microsoft.Azure.Amqp.Framing;
+using Microsoft.Azure.Cosmos;
+using Microsoft.Azure.Cosmos.Linq;
+using Microsoft.Extensions.Logging;
+using NCS.DSS.Customer.Models;
+using System.Net;
+using System.Text.Json;
+using Container = Microsoft.Azure.Cosmos.Container;
+
+namespace NCS.DSS.Customer.Cosmos.Provider
+{
+    public class CosmosDBProvider : ICosmosDBProvider
+    {
+        private readonly Container _container;
+        private readonly string _databaseId = Environment.GetEnvironmentVariable("DatabaseId");
+        private readonly string _containerId = Environment.GetEnvironmentVariable("CollectionId");
+        private readonly ILogger<CosmosDBProvider> _logger;
+        public CosmosDBProvider(CosmosClient cosmosClient,ILogger<CosmosDBProvider> logger)
+        {
+            _container = cosmosClient.GetContainer(_databaseId, _containerId);
+            _logger = logger;
+        }
+        public async Task<bool> DoesCustomerResourceExist(Guid customerId)
+        {
+            try
+            {
+                var queryCust = _container.GetItemLinqQueryable<Models.Customer>().Where(x => x.CustomerId == customerId).ToFeedIterator();
+
+                while (queryCust.HasMoreResults)
+                {
+                    var response = await queryCust.ReadNextAsync();
+                    if (response != null)
+                    {
+                        _logger.LogInformation("Customer Record found in Cosmos DB for {CustomerID}", customerId);
+                        return true;
+                    }                    
+                }
+                _logger.LogWarning("No Customer Record found with {CustomerID} in Cosmos DB", customerId);
+                return false;
+            }
+            catch (CosmosException ce)
+            {
+                _logger.LogError("Failed to find the Customer Record in Cosmos DB {CustomerID}. Exception {Exception}.", customerId, ce.Message);
```

Comment:
The log should pass the exception object:
`_logger.LogError(ce, "Failed to find the Customer Record in Cosmos DB {CustomerID}. Exception {Exception}.", customerId, ce.Message);`